src/coreclr/jit/importercalls.cpp

```diff
@@ -8118,18 +8118,18 @@ GenTree* Compiler::impMinMaxIntrinsic(CORINFO_METHOD_HANDLE method,
 
                 if (isNumber)
                 {
-                    needsFixup = cnsNode->IsFloatNegativeZero();
+                    needsFixup = cnsNode->IsFloatPositiveZero();
                 }
                 else
                 {
-                    needsFixup = cnsNode->IsFloatPositiveZero();
+                    needsFixup = cnsNode->IsFloatNegativeZero();
                 }
```

Comment:
We weren't doing the wrong thing, just not being as efficient as we should've been.